pinot-controller/src/main/java/org/apache/pinot/controller/ControllerConf.java

```diff
@@ -133,6 +134,7 @@ private static long getRandomInitialDelayInSeconds() {
   // If it's set to false, existing HLC realtime tables will stop consumption, and creation of new HLC tables will be disallowed.
   // Please make sure there is no HLC table running in the cluster before disallowing it.
   private static final String ALLOW_HLC_TABLES = "controller.allow.hlc.tables";
+  private static final String HELIX_INSTANCE_TYPE = "controller.helix.instance.type";
```

Comment:
let us not use a config to make sure that only controllers register as participants. Instead, let us have two constructors in PinotHelixResourceManager: One with (config) and the other with (config, instanceType). We use the first ctor everywhere except the ControllerStarter. In the controllerstarter, we can either use the second ctor, or use some static method like makePnotHelixResoirceManagerForController() and make the second ctor private.

Reply:
Done.